WORKSPACE

```diff
@@ -308,14 +308,18 @@ rules_haskell_asterius_toolchains(
 )
 
 load(
-    "@rules_haskell//haskell:ghc_bindist.bzl",
-    "haskell_register_ghc_bindists",
+    "@rules_haskell//haskell:ghc_bindist_hadrian.bzl",
+    "haskell_register_ghc_bindists_hadrian",
 )
 
-haskell_register_ghc_bindists(
+haskell_register_ghc_bindists_hadrian(
+    url = "file:///home/guillaume/ExternalPrograms/ghc/_build/bindist/ghc-9.2.5-x86_64-unknown-linux.tar.xz",
+    sha256 = "e27724de38998dd6c3fb46ac5df4cf6818d779c1b749ea1afcd0b64e55b8217e",
+    strip_prefix = "ghc-9.2.5-x86_64-unknown-linux",
+    version = "9.2.5",
```

Comment:
Some of the issues are probably unrelated to the hadrian toolchain and instead just due to the version bump. It might make sense to tackle those in a separate PR and [bump the GHC version under test](https://github.com/tweag/rules_haskell/blob/master/constants.bzl#L1).

Reply:
You are completely right. Separating issues is always beneficial and it will help both review and understanding the cause of the failing tests.